internal/k8s/deployments_test.go

```diff
@@ -557,3 +566,238 @@ func TestComputeConnectionSpecHash(t *testing.T) {
+			for _, unexpectedKey := range tt.unexpectedEnvVars {
+				value, exists := envMap[unexpectedKey]
+				assert.False(t, exists, "Environment variable %s should not be present, but found value: %s", unexpectedKey, value)
+			}
+
+			// For mTLS case, verify volume mounts and volumes are configured
+			if tt.connection.MutualTLSSecret != "" {
+				assert.Len(t, container.VolumeMounts, 1)
+				assert.Equal(t, "temporal-tls", container.VolumeMounts[0].Name)
+				assert.Equal(t, "/etc/temporal/tls", container.VolumeMounts[0].MountPath)
+
+				assert.Len(t, deployment.Spec.Template.Spec.Volumes, 1)
+				assert.Equal(t, "temporal-tls", deployment.Spec.Template.Spec.Volumes[0].Name)
+				assert.Equal(t, tt.connection.MutualTLSSecret, deployment.Spec.Template.Spec.Volumes[0].VolumeSource.Secret.SecretName)
+			}
+		})
+	}
+}
+
+func TestNewDeploymentWithOwnerRef_EnvConfigSDKCompatibility(t *testing.T) {
+	// Test that the environment variables injected by the controller
+	// can be parsed by the official Temporal SDK envconfig package
+
+	tests := map[string]struct {
+		connection      temporaliov1alpha1.TemporalConnectionSpec
+		expectedAddress string
+		expectTLS       bool
+	}{
+		"without TLS": {
+			connection: temporaliov1alpha1.TemporalConnectionSpec{
+				HostPort: "test.temporal.example:9999",
+			},
+			expectedAddress: "test.temporal.example:9999",
+			expectTLS:       false,
+		},
+		"with TLS": {
+			connection: temporaliov1alpha1.TemporalConnectionSpec{
+				HostPort:        "mtls.temporal.example:8888",
+				MutualTLSSecret: "test-tls-secret",
+			},
+			expectedAddress: "mtls.temporal.example:8888",
+			expectTLS:       true,
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			spec := &temporaliov1alpha1.TemporalWorkerDeploymentSpec{
+				Template: corev1.PodTemplateSpec{
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name:  "worker",
+								Image: "temporal/worker:latest",
+							},
+						},
+					},
+				},
+				WorkerOptions: temporaliov1alpha1.WorkerOptions{
+					TemporalNamespace: "test-namespace",
+				},
+			}
+
+			deployment := k8s.NewDeploymentWithOwnerRef(
+				&metav1.TypeMeta{},
+				&metav1.ObjectMeta{Name: "test", Namespace: "default"},
+				spec,
+				"test-deployment",
+				"test-build-id",
+				tt.connection,
+			)
+
+			// Extract environment variables from the deployment
+			container := deployment.Spec.Template.Spec.Containers[0]
+
+			// Set environment variables using t.Setenv() to simulate the runtime environment
+			for _, env := range container.Env {
+				t.Setenv(env.Name, env.Value)
+			}
+
+			if tt.expectTLS {
+				// Create temporary test certificate files
+				tempDir := t.TempDir()
+				certPath := filepath.Join(tempDir, "client.pem")
+				keyPath := filepath.Join(tempDir, "client.key")
+
+				// Generate a self-signed certificate for testing
+				privateKey, err := rsa.GenerateKey(rand.Reader, 2048)
```

Comment:
Is there a lighter weight algo we could use for testing? Add a godoc comment exercising caution to avoid copy/paste if there is but it's less secure.